d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerTest.java

```diff
@@ -427,6 +429,177 @@ public void testClusterInfoProviderUnregisterClusterListener()
     Assert.assertEquals(testClusterListener.getClusterRemovedCount(CLUSTER1_NAME), 0, "expected unchanged remove count of 0 because unregistered");
   }
 
+  @Test
+  public void testListenToServiceAndClusterTimeout() throws ExecutionException, InterruptedException,
+          ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IllegalAccessException
+  {
```

Comment:
This will probably make your life easier:
```suggestion
  @Test
  public void testListenToServiceAndClusterTimeout() throws Exception
  {
```

Reply:
Done